fifteen_min/src/isochrone.rs

```diff
@@ -113,9 +120,36 @@ impl Isochrone {
             return None;
         }
 
+        let from = if self.start.len() > 1 {
+            let b_hash_map: HashMap<BuildingID, Duration> = self
```

Comment:
Hovering over buildings in the amenity search mode is still slow on my machine. This is the reason why. This strategy works like this:

1) Do a Dijkstra's floodfill from the building being hovered on, stopping after 15 minutes away from that building
2) Find the closest store
3) Calculate the one path

The floodfill is actually just slow. I think we should be able to get a speedup by:

1) Calculating the path between `to` and each of the buildings in `self.start`
  2) Choosing the best path from those. The lazy approach is `path.total_length()`, which goes by distance... And by the way, [min_by_key](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.min_by_key) is a convenient way to do that calculation.

It seems counter-intuitive that doing `n` pathfinding queries is faster than one floodfill from `n` points, but pathfinding is very fast because it uses [contraction hierarchies](https://en.wikipedia.org/wiki/Contraction_hierarchies) internally. We should of course try it out to be sure, but I'm guessing this'll work.

The disadvantage of using `pathfind` instead of `times_from_buildings` is that the walking/biking options won't be used. The contraction hierarchies are built just once for each mode, with default speeds set. I think it's fine here, because we're just aiming to show the path to what's most likely the closest store. The path might deviate a little based on walking options configured in the future (like avoiding steep elevation). But even in the main viewer, we're calling `pathfind` and just assuming these default options. So it's fine.

Reply:
Oh yeah, and on native, we can speed this up even more by using `timer.parallelize` on `self.start`. Calculate all of the paths simultaneously. On web, this will automatically just use the single thread, since that's all that's available. No need to wrap this in `ctx.loading_screen` -- that would be very noisy to the user as they move the mouse around! You can just do `Timer::throwaway().parallelize()` -- that'll avoid spamming the terminal with logs about how long things took.

Reply:
Yes, this seems like a good approach, I didn't fully appreciate how much faster the pathfinding was. I took at stab at it in my last commit, running into a rust problem but overall the approach is much cleaner and it should work for 1 start point or multiple. Once this is working I can work on parallelizing.